alliance/apps/backlog/views/list_backlogs_view.py

```diff
@@ -151,35 +152,31 @@ def select_sprint(self, request, results):
                     str(e), code="exception")
 
     def update_backlog_and_acc_cri(self, request, results):
-        from ...backlog.forms import BacklogUpdateForm
-        backlog_id = request.POST.get('backlog-id')
-        try:
-            backlog = Backlog.objects.get(id=backlog_id)
-        except Backlog.DoesNotExist:
-            backlog = None
-        else:
-            form = BacklogUpdateForm(request.POST,
-                                     prefix='backlog', instance=backlog)
+        from ...backlog.forms import AcceptanceCriteriaFormSet, BacklogUpdateForm
 
-            prefix = 'acceptance-criteria-%d' % backlog.id
-            from ...backlog.forms import AcceptanceCriteriaFormSet
-            formset = AcceptanceCriteriaFormSet(request.POST,
-                                                instance=backlog,
-                                                prefix=prefix)
-            if form.is_valid():
-                if formset.is_valid():
-                    backlog = form.save()
-                    backlog.refresh_from_db()
-                    formset.save()
-                    formset = AcceptanceCriteriaFormSet(
+        backlog_id = request.POST.get('backlog-id')
+        backlog = Backlog.objects.get_object_or_none(id=backlog_id)
```

Comment:
Good catches, Brad! I'll go ahead and make the changes. I can't wait until
we have tests for these things. It'll make development so much simpler.

Cheers,
h

On Wed, Oct 7, 2015 at 10:10 AM, B Martsberger <notifications@github.com>
wrote:

> In alliance/apps/backlog/views/list_backlogs_view.py
> <https://github.com/NorthBridge/alliance-community/pull/31#discussion_r41402261>
> :
>
> >
> > -            prefix = 'acceptance-criteria-%d' % backlog.id
> > -            from ...backlog.forms import AcceptanceCriteriaFormSet
> > -            formset = AcceptanceCriteriaFormSet(request.POST,
> > -                                                instance=backlog,
> > -                                                prefix=prefix)
> > -            if form.is_valid():
> > -                if formset.is_valid():
> > -                    backlog = form.save()
> > -                    backlog.refresh_from_db()
> > -                    formset.save()
> > -                    formset = AcceptanceCriteriaFormSet(
> > +        backlog_id = request.POST.get('backlog-id')
> > +        backlog = Backlog.objects.get_object_or_none(id=backlog_id)
>
> Here too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NorthBridge/alliance-community/pull/31/files#r41402261>
> .
>